DisCatSharp.ApplicationCommands/Workers/ApplicationCommandWorker.cs

```diff
@@ -0,0 +1,383 @@
+								));
+							}
+						}
+
+						subNameLocalizations = subCommandTranslation.NameTranslations;
+						subDescriptionLocalizations = subCommandTranslation.DescriptionTranslations;
+					}
+
+
+					//Creates the subcommand and adds it to the main command
+					var subpayload = new DiscordApplicationCommandOption(commandAttribute.Name, commandAttribute.Description, ApplicationCommandOptionType.SubCommand, null, null, localizisedOptions ?? options, nameLocalizations: subNameLocalizations, descriptionLocalizations: subDescriptionLocalizations);
+					payload = new DiscordApplicationCommand(payload.Name, payload.Description, payload.Options?.Append(subpayload) ?? new[] { subpayload }, payload.DefaultPermission, nameLocalizations: payload.NameLocalizations, descriptionLocalizations: payload.DescriptionLocalizations);
+					commandTypeSources.Add(new KeyValuePair<Type, Type>(subclassinfo, type));
+
+					//Adds it to the method lists
+					commandmethods.Add(new KeyValuePair<string, MethodInfo>(commandAttribute.Name, submethod));
+					groupCommands.Add(new GroupCommand { Name = groupAttribute.Name, Methods = commandmethods });
+				}
+
+				var command = new SubGroupCommand { Name = groupAttribute.Name };
+				//Handles subgroups
+				foreach (var subclass in subclasses)
+				{
+					var subGroupAttribute = subclass.GetCustomAttribute<SlashCommandGroupAttribute>();
+					var subsubmethods = subclass.DeclaredMethods.Where(x => x.GetCustomAttribute<SlashCommandAttribute>() != null);
+
+					var options = new List<DiscordApplicationCommandOption>();
+
+					var currentMethods = new List<KeyValuePair<string, MethodInfo>>();
+
+
+					DiscordApplicationCommandLocalization subNameLocalizations = null;
+					DiscordApplicationCommandLocalization subDescriptionLocalizations = null;
+
+					if (translator != null)
+					{
+						var commandTranslation = translator.Single(c => c.Name == payload.Name);
+						if (commandTranslation != null && commandTranslation.SubGroups != null)
+						{
+
+							var subCommandTranslation = commandTranslation.SubGroups.Single(sc => sc.Name == subGroupAttribute.Name);
+
+							if (subCommandTranslation != null)
+							{
+								subNameLocalizations = subCommandTranslation.NameTranslations;
+								subDescriptionLocalizations = subCommandTranslation.DescriptionTranslations;
+							}
+						}
+					}
+
+					//Similar to the one for regular groups
+					foreach (var subsubmethod in subsubmethods)
+					{
+						var suboptions = new List<DiscordApplicationCommandOption>();
+						var commatt = subsubmethod.GetCustomAttribute<SlashCommandAttribute>();
+						var parameters = subsubmethod.GetParameters();
+						if (parameters.Length == 0 || parameters == null || !ReferenceEquals(parameters.First().ParameterType, typeof(InteractionContext)))
+							throw new ArgumentException($"The first argument must be an InteractionContext!");
+
+						parameters = parameters.Skip(1).ToArray();
```

Comment:
Skipping over the first parameter breaks commands like below as it skips over the first option for `test_command` and `reverse_test`.
```cs
[SlashCommandGroup("top_level", "top level group")]
public class MySlashCommandGroups : ApplicationCommandsModule
{
    [SlashCommandGroup("mid_level", "mid level group")]
    public class MidLevelGroup : ApplicationCommandsModule
    {
        [SlashCommand("test_command", "test commands")]
        public async Task TestCommand(InteractionContext ctx,
            [Option("repeat_this", "repeats the text")] string repeatMe)
        {
            await ctx.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource, new DiscordInteractionResponseBuilder()
                .WithContent(repeatMe));
        }
    }

    [SlashCommandGroup("mid_level_two", "mid level group two")]
    public class MidLevelGroupTwo : ApplicationCommandsModule
    {
        [SlashCommand("reverse_test", "test command to reverse a string")]
        public async Task TestReverseCommand(InteractionContext ctx,
            [Option("reverse_this", "reverses the text")] string reverseMe)
        {
            char[] charArray = reverseMe.ToCharArray();
            Array.Reverse( charArray );
            await ctx.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource,
                new DiscordInteractionResponseBuilder()
                    .WithContent(new string(charArray)));
        } 
    }
}
```

Reply:
Something else happening here actually with options. This would be better though than skipping 1
```cs
parameters = parameters.SkipWhile(p => ReferenceEquals(p.ParameterType, typeof(InteractionContext)) ).ToArray();
```

Reply:
A skip while makes it sound like we have more than 1 interaction context in our method signature which should never be the case. Also, the condition on line above checks the first parameter having the type of InteractionContext so doing a Skip(1) is better / cleaner in this scenario.

However, I suppose if we wanted to support throwing the interaction context anywhere in the method signature we'd want to take another approach.